server/src/main/java/org/elasticsearch/search/retriever/KnnRetrieverBuilder.java

```diff
@@ -73,6 +75,9 @@ public final class KnnRetrieverBuilder extends RetrieverBuilder<KnnRetrieverBuil
     }
 
     public static KnnRetrieverBuilder fromXContent(XContentParser parser, RetrieverParserContext context) throws IOException {
+        if (context.clusterSupportsFeature(NODE_FEATURE) == false) {
+            throw new IllegalArgumentException("[" + NAME + "] retriever is not a supported feature");
```

Comment:
I have a weak preference for making this error message consistent with the retriever not existing at all. That is, it should appear to the user that this retriever doesn't exist, just like it would have before we got into a mixed cluster scenario that we are in here.

Reply:
Agreed! I will change these and add an additional check for the general retriever key as well.